unskript-ctl/add_notification.py

```diff
@@ -0,0 +1,170 @@
+        parser.add_argument('-u',
+                            '--web-hook-url',
+                            required=True,
+                            help="Slack Notification Webhook URL")
+        parser.add_argument('-c',
+                            '--channel-name',
+                            required=True,
+                            help="Channel Name to send Notification")
+        args = parser.parse_args(sys.argv[3:])
+        if len(sys.argv[3:]) != 4:
+            parser.print_help()
+            sys.exit(0)
+        
+        data = {'hook_url' : args.web_hook_url, 'channel': args.channel_name}
+        unskript_ctl_config_create_notification('slack', data) 
+    
+    def SMTP(self):
+        """SMTP: This function implements the sub-parser for Mail/SMTP Notification
+        """
+        parser = ArgumentParser(description="Add Mail Notification")
+        parser.add_argument('-s',
+                            '--smtp-host',
+                            required=True,
+                            help="SMTP Hostname like smtp.google.com")
+        parser.add_argument('-u',
+                            '--smtp-user',
+                            required=True,
+                            help="SMTP Username like user@domain.com")
+        parser.add_argument('-p',
+                            '--smtp-password',
+                            required=True,
+                            help="SMTP Password for the above user")
+        parser.add_argument('-t',
+                            '--to-email',
+                            required=True,
+                            help="Reciever Email address like reciever@example.com")
+        args = parser.parse_args(sys.argv[3:])
+        if len(sys.argv[3:]) != 8:
+            parser.print_help()
+            sys.exit(0)
+        
+        data = {'smtp_host' : args.smtp_host, 
+                'smtp_user': args.smtp_user,
+                'smtp_password': args.smtp_password,
+                'to_email': args.to_email}
+        unskript_ctl_config_create_notification('mail', data) 
+
+    def delete_notification(self, type):
+        """Implementing Delete of the cruD for Notification
+        """
+        if not type:
+            return 
+        
+        if type.lower() == 'slack':
+            unskript_ctl_config_delete_notification('slack')
+        elif type.lower() == 'smtp':
+            unskript_ctl_config_delete_notification('mail')
+        else:
+            print("ERROR: Wrong choice")
+    
+    def update_notification(self, type):
+        """update_notification: Implementing cUrd for Notification
+        """
+        if not type:
+            return 
+        if type.lower() == 'slack':
+            self.SLACK()
+        elif type.lower() == 'smtp':
+            self.SMTP()
+        else:
+            print("ERROR: Wrong choice")
+
+    def read_notification(self, type):
+        """read_notification: Implementing cuRd for Notification
```

Comment:
This is only implementing read, not curd?? (crud)

Reply:
:) Spelling and Me!